Source/x11/XGServerWindow.m

```diff
@@ -4409,75 +4415,152 @@ - (void) freecursor: (void*) cid
+                  }
+            }
+        }
+      XRRFreeScreenResources(screen_res);
+      screens = [NSArray arrayWithArray: tmpScreens];
+    }
+#endif
+
+  if (monitorsCount == 0)
+    {
+      /* Assumed that it's always available 1 screen per application. We only
+         need to know it's number and it was saved in _initXContext as
+         `defScreen`. */
+      monitorsCount = 1;
+      monitors = NSZoneMalloc([self zone], sizeof(MonitorDevice));
+      monitors[0].screen_id = defScreen;
+      monitors[0].depth = [self windowDepthForScreen: defScreen];
+      monitors[0].resolution = [self resolutionForScreen: defScreen];
+      monitors[0].frame = NSMakeRect(0, 0,
+                                     DisplayWidth(dpy, defScreen),
+                                     DisplayHeight(dpy, defScreen));
+      screens = [NSArray arrayWithObject: [NSNumber numberWithInt: defScreen]];
+    }
+  
+  return screens;
 }
 
-- (NSWindowDepth) windowDepthForScreen: (int) screen_num
+// `screen` is a monitor index not X11 screen
+- (NSWindowDepth) windowDepthForScreen: (int)screen
 {
-  Screen	*screen;
+  Screen	*x_screen;
   int		 class = 0, bpp = 0;
 
-  screen = XScreenOfDisplay(dpy, screen_num);
-  if (screen == NULL)
+  if (screen < 0 || screen >= monitorsCount)
+    {
+      return 0;
+    }
+  
+  x_screen = XScreenOfDisplay(dpy, monitors[screen].screen_id);
+  
+  if (x_screen == NULL)
     {
       return 0;
     }
 
-  bpp = screen->root_depth;
-  class = screen->root_visual->class;
+  bpp = x_screen->root_depth;
+  class = x_screen->root_visual->class;
 
   return _computeDepth(class, bpp);
 }
 
-- (const NSWindowDepth *) availableDepthsForScreen: (int) screen_num
+// `screen` is a monitor index not X11 screen
+- (const NSWindowDepth *) availableDepthsForScreen: (int)screen
 {
-  Screen	*screen;
+  Screen	*x_screen;
   int		 class = 0;
   int		 index = 0;
   int		 ndepths = 0;
   NSZone	*defaultZone = NSDefaultMallocZone();
   NSWindowDepth	*depths = 0;
 
-  if (dpy == NULL)
+  if (dpy == NULL || screen < 0 || screen >= monitorsCount)
     {
       return NULL;
     }
 
-  screen = XScreenOfDisplay(dpy, screen_num);
-  if (screen == NULL)
+  x_screen = XScreenOfDisplay(dpy, monitors[screen].screen_id);
+  
+  if (x_screen == NULL)
     {
       return NULL;
     }
 
   // Allocate the memory for the array and fill it in.
-  ndepths = screen->ndepths;
-  class = screen->root_visual->class;
+  ndepths = x_screen->ndepths;
+  class = x_screen->root_visual->class;
   depths = NSZoneMalloc(defaultZone, sizeof(NSWindowDepth)*(ndepths + 1));
   for (index = 0; index < ndepths; index++)
     {
-      int depth = screen->depths[index].depth;
+      int depth = x_screen->depths[index].depth;
       depths[index] = _computeDepth(class, depth);
     }
   depths[index] = 0; // terminate with a zero.
 
   return depths;
 }
 
-- (NSSize) resolutionForScreen: (int)screen_num
+// `screen` is a Xlib screen number.
```

Comment:
Wouldn't the name screen_num be better in that case?

Reply:
Agree. Thanks, fixed.

Reply:
Now the commented out code uses the wring variable name. But no harm in that.

Reply:
For the sake of perfection I've fixed it. :-)